sdks/python/apache_beam/runners/worker/sideinputs.py

```diff
@@ -68,9 +73,29 @@ def __init__(self, sources,
     # Whether an error was encountered in any source reader.
     self.has_errored = False
 
+    self.read_counter = read_counter or opcounters.TransformIOCounter()
+
     self.reader_threads = []
     self._start_reader_threads()
 
+  def add_byte_counter(self, reader):
+    """Adds byte counter observer to a side input reader.
+
+    If the 'sideinput_io_metrics' experiment flag is not passed in, then nothing
+    is attached to the reader.
+
+    Args:
+      reader: A reader that should inherit from ObservableMixin to have
+        bytes tracked.
+    """
+    def update_bytes_read(record_size, is_record_size=False, **kwargs):
+      # Let the reader report block size.
+      if is_record_size:
+        self.read_counter.add_bytes_read(record_size)
+
+    if self.read_counter and isinstance(reader, observable.ObservableMixin):
```

Comment:
Would `self.read_counter` be ever None? Is this a necessary check?

Is it an error if the reader is not a instance of ObservableMixin?